airwallex/src/main/java/com/airwallex/android/view/PaymentMethodsActivity.kt

```diff
@@ -158,9 +168,30 @@ class PaymentMethodsActivity : AirwallexCheckoutBaseActivity() {
                     onSuccess = {
                         setLoadingProgress(loading = false)
                         // Complete load available payment method type
-                        this@PaymentMethodsActivity.availablePaymentMethodTypes = it
-                        initView(it)
-                        filterPaymentConsents()
+                        val availableMethodTypes = it
+                        val cardPaymentMethod = it.find { it.name == PaymentMethodType.CARD.value }
+
+                        val availablePaymentConsents = filterPaymentConsents(cardPaymentMethod)
+
+                        this@PaymentMethodsActivity.availablePaymentMethodTypes = availableMethodTypes
+                        this@PaymentMethodsActivity.availablePaymentConsents = availablePaymentConsents
+
+                        // skip straight to the individual card screen?
+                        val hasPaymentConsents = availablePaymentConsents.isNotEmpty()
+                        val availablePaymentMethodsSize = availableMethodTypes.size
+
+                        val hasSinglePaymentMethod = !hasPaymentConsents && availablePaymentMethodsSize == 1
+
+                        if (cardPaymentMethod != null && hasSinglePaymentMethod) {
```

Comment:
This is correct, but could we extract this logic to this activity's view model `PaymentMethodsViewModel` to make it testable?